File: Ruby/ruby_squares/get_squares.rb

require 'pry'

def get_squares (array)
  squareArray = []
  array.each_index  do |index|
    if array[index] ** 0.5 == (array[index] ** 0.5).round
      squareArray.push array[index]
    end
  end
  squareArray.sort.uniq
end

# get_squares([4, 1, 16, 1, 10, 35, 22])

binding.pry

File: Ruby/ruby_squares/getSquares.js

var getSquares = function (array) {
  var squareArray = [];
  for (var index = 0; index < array.length; index++) {
    if (Math.sqrt(array[index]) == Math.round(Math.sqrt(array[index])))
      squareArray.push(array[index]);
  }
  console.log (squareArray);


  squareArray.sort(function(a, b) {
    return a - b;
  });
  //
  // squareArray.map ()
};
